### DISCUSSION TITLE
How to test that a tooltip is shown?

### DISCUSSION THREAD

**Discussion by JoosepAlviste** [Akryum/v-tooltip#921] [Q&A]
Hey, thanks for the library!

I'm trying to write a unit test that checks if a tooltip is shown (using the `v-tooltip` directive) but can't quite get it to show up. I'm using Vue 3, `floating-vue` v2.0.0-beta.20, Jest and Vue Testing Library. I just have a simple component like this:

```vue
<template>
	<h1 v-tooltip="'Foobar'">Hello World</h1>
</template>

<script lang="ts" setup>
</script>
```

And my test is also very simple:

```vue
import { render, screen, fireEvent } from '@testing-library/vue';
import MyComponent from './MyComponent.vue';

it('shows a tooltip', async () => {
	render(MyComponent);

	await fireEvent.mouseEnter(screen.getByText('Hello World'));

	screen.getByText('Foobar');
});
```

Getting the tooltip by text fails and the element also does not exist in the debug output of the HTML:

```
    <body>
      <div>
        <h1
          class="v-popper--has-tooltip"
        >
          Hello World
        </h1>
      </div>
      <div
        data-v-app=""
      >
        
        
        
        
      </div>
    </body>
```

I also tried a few other events but couldn't find one that worked. If I recall correctly, triggering the tooltip in a test like this used to work with `v-tooltip`.

Is there a specific way to test this?

**Comment by furtivesock** [comment]
Bump, I'm having the same issue using vue-test-utils, I'm clueless tbh

**Reply by JoosepAlviste** [reply]
Yeah, it would be great to have some clarity on this since testing is such an important part of software development 😕

**Reply by furtivesock** [reply]
Hi! So I've found that using `requestAnimationFrame` method ([MDN docs](https://developer.mozilla.org/fr/docs/Web/API/Window/requestAnimationFrame)) after firing the event may help DOM to get modified (in my case I was testing a `Dropdown` component).

Two ways to do so:
- Using a promise (+ Jest's `done` callback) : 
```js
// Do the trigger
requestAnimationFrame(() => {
  // Make assertions
  done()
})

```
- Using `async/await` : 
```js
// Do the trigger
await new Promise(requestAnimationFrame)
// Make assertions
``` 

Many thanks to this reply: https://github.com/vuejs/vue-test-utils/issues/1421#issuecomment-581102508
